### ISSUE TITLE
bug:  Make Spring Security dependency optional in the graphql-dgs

### ISSUE THREAD

**Issue by hantsy** [Netflix/dgs-framework#631]
Tried to create a webflux example project, I did not add Spring Security in the dependencies, and but I found the Spring Security in the dependency tree. 

I found the `spring-security-core` dep in the `graphql-dgs` module,  https://github.com/Netflix/dgs-framework/blob/master/graphql-dgs/build.gradle.kts#L33

**Comment by berngp** [comment]
@hantsy could you help us understand why `spring-securit-core` was problematic in this particular case?

**Comment by hantsy** [comment]
In my [Dgs WebFlux example](https://github.com/hantsy/spring-graphql-sample/blob/java17/dgs-webflux/pom.xml#L57-L64), I do not want to add security at the moment , but they are in the final build, I have to exclude them manually. It is not a big issue, but I think we could have to make the dependencies clean as possible.

**Comment by paulbakker** [comment]
I agree with @hantsy. If the dependency isn't required, it shouldn't be pulled in.

The only dependency I _know_ of is the Exception check in the exception handler, but it uses this to account for the decency not being there: 

```
val springSecurityAvailable = try {
            Class.forName("org.springframework.security.access.AccessDeniedException")
            true
        } catch (ex: ClassNotFoundException) {
            false
        }
```

We'll have to test if there are other parts where it's expected to be available.

**Comment by berngp** [comment]
@paulbakker I also agree that the dep shouldn't be _forced_. There are plenty of other Security Frameworks that could replace Spring Security successfully. We should be aware that if we remove this in a minor version there would be builds that make break since they assume they will get it transitively.

**Comment by hantsy** [comment]
@paulbakker The spring-core module includes a [`ClassUtil.isPresent` to detect the existence of a class](registry.aliyuncs.com/google_containers).

**Comment by paulbakker** [comment]
> @paulbakker I also agree that the dep shouldn't be _forced_. There are plenty of other Security Frameworks that could replace Spring Security successfully. We should be aware that if we remove this in a minor version there would be builds that make break since they assume they will get it transitively.

It shouldn't, because it was an `implementation` dependency, not `api`.

**Comment by berngp** [comment]
Addressed in v4.8.0.
